x/authz/authorizations.go

```diff
@@ -0,0 +1,38 @@
+package authz
+
+import (
+	"github.com/gogo/protobuf/proto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Authorization represents the interface of various Authorization types implemented
+// by other modules.
+type Authorization interface {
+	proto.Message
+
+	// MsgTypeURL returns the fully-qualified Msg service method URL (as described in ADR 031),
+	// which will process and accept or reject a request.
+	MsgTypeURL() string
```

Comment:
@robert-zaremba You mentioned this might conflict with https://github.com/cosmos/cosmos-sdk/pull/9042#discussion_r621311724, right?

Maybe an idea here is `GetMsgTypeURL`?

Reply:
In `GenericAuthorization` I have `msg` - I like the short name.  But if you prefer, I can change it and use `Get...` here.

Reply:
Also, `Get...` is not very go style.